alibi/explainers/permutation_importance.py

```diff
@@ -558,39 +558,43 @@ def _compute_exact(self,
         A dictionary having as keys the metric names and as values the permutation feature importance associated
         with the corresponding metrics.
         """
-        y_hat = []
+        y_perm, y_perm_hat = [], []
         weights: Optional[List[np.ndarray]] = [] if sample_weight else None
 
         for i in range(len(X)):
-            # create dataset
-            X_tmp = np.tile(X[i:i+1], reps=(len(X) - 1, 1))
+            # create input features dataset
+            X_tmp = np.tile(X[i:i+1], reps=(len(X) - 1, ) + (1, ) * (len(X.shape) - 1))
             X_tmp[:, features] = np.delete(arr=X[:, features], obj=i, axis=0)
 
+            # create ground-truth labels
+            y_tmp = np.tile(y[i:i+1], reps=(len(y) - 1, ) + (1, ) * (len(y.shape) - 1))
```

Comment:
Is it worth adding shapes in comments for these arrays? Otherwise hard to see what's going on. Would also appreciate a short explanation of the failure case for the previous implementation (an example of multi-dimensional ground truth).